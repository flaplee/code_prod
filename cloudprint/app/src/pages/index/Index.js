import React, { Component } from 'react'
import { render } from 'react-dom'
import Cookies from 'react-cookies';
import 'whatwg-fetch'
import {
    BrowserRouter as Router,
    Route
} from 'react-router-dom'

import Nav from './Nav'
import Printer from './Printer'
import { serverIp, path, baseURL, mpURL, convertURL, timeout, mockURL } from '../../configs/config'
class PrintIndex extends Component{
    constructor(props) {
        super(props);
        this.state = {
            user:{
                user_id: '',
                org_id: '',
                token: ''
            },
            sn: '' || props.location.search.sn,
            printer:{},
            printerList:{},
            inkbox:{},
            navlist: [{
                title: '选择图片打印',
                url: ''
            },
            {
                title: '选择文件打印',
                url: ''
            },
            {
                title: '打印任务',
                url: ''
            },
            {
                title: '打印机管理',
                url: ''
            }]
        };
    };

    componentWillUpdate(props, state) {
        
    }

    componentWillMount(){}
    
    componentDidMount() {
        const self = this
        if (self.state.sn != '') { Cookies.save('sn', self.state.sn, { path: '/' });}
        let signPrint;
        let appIdPrint;
        let timestampPrint = '1536041270823';
        let nonceStrPrint = 'abcdefg';
        if (Cookies.load('sign') && Cookies.load('timestamp') && Cookies.load('nonceStr')) {
            signPrint = Cookies.load('sign')
            timestampPrint = Cookies.load('timestamp')
            nonceStrPrint = Cookies.load('nonceStr')
        } else {
            timestampPrint = (Date.now().toString())
            nonceStrPrint = "abcdefg";
        }

        self.getAppData(timestampPrint, nonceStrPrint, function (response) {
            if (response && response.signStr) {
                signPrint = response.signStr
                appIdPrint = response.appId
                Cookies.save('sign', signPrint, { path: '/' });
                Cookies.save('appId', appIdPrint, { path: '/' });
                Cookies.save('timestamp', timestampPrint, { path: '/' });
                Cookies.save('nonceStr', nonceStrPrint, { path: '/' });
            }

            // 注入配置信息
            deli.config({
                noncestr: nonceStrPrint, // 必填，生成签名的随机串
                appId: appIdPrint || Cookies.load('appId'), // 必填，应用ID  373175764691976192
                timestamp: timestampPrint, // 必填，生成签名的时间戳
                signature: signPrint || Cookies.load('sign') // 必填，服务端生成的签名 26fcd1cab8ff455bfea0ee59a67bf122
            });

            // 验证签名成功
            deli.ready(function () {
                // 通用设置
                deli.common.navigation.setTitle({
                    "title": "得力云打印"
                }, function(data) {}, function(resp) {});

                deli.common.navigation.setRight({
                    "icon": "http://t.static.delicloud.com/h5/web/cloudprint/images/icon/scan_code.png"
                }, function(data) {
                    deli.app.code.scan({
                        app_id: ''
                    }, function (json) {
                        alert(JSON.stringify(json))
                    }, function(resp) {});
                }, function(resp) {});

                deli.common.navigation.setColors({
                    "color": "#5D85E0",
                    "notify_theme": "white",
                    "nav_theme": "white"
                }, function (data) {}, function (resp) {});

                // 关闭
                deli.common.navigation.close({}, function (data) {
                    // 重置设置
                }, function (resp) {});

                deli.app.user.get({
                    "user_id": ""
                }, function (data) {
                    if (data) {
                        self.setState({ "user": { "user_id": data.user.id } })
                        deli.app.organization.get({
                            "org_id": ""
                        }, function (odata) {
                            self.setState({ "user": { "org_id": odata.organization.id } })
                            deli.app.session.get({
                                user_id: data.user.id
                            }, function (udata) {
                                self.getLocalData({
                                    user_id: data.user.id,
                                    org_id: odata.organization.id,
                                    token: udata.token
                                }, self.state.sn);

                            }, function (uresp) {
                            });
                        }, function (oresp) {
                        });
                    }
                }, function (resp) {
                });
            });

            //验证签名失败
            deli.error(function () {
                console.log("deli.error")
            });

            // 调试
            /* self.setState({ "user": { "token": Cookies.load('token') } })
            self.getLocalData({
                user_id: Cookies.load('userId'),
                org_id: Cookies.load('orgId'),
                token: Cookies.load('loginToken')
            }); */
        });
    }
    
    //监听Printer变化状态
    transPrinter(printer) {
        console.log("printer", printer);
    }
    
    renderPrinter() {
        const user = this.state.user;
        const printer = this.state.printer;
        const result = <Printer user={user} printer={printer}></Printer>;
        return result;
    }

    //获取auth信息
    getLocalData(data, printSn){
        var self = this
        //登录api
        fetch(mpURL + '/a/auth/login', {
            method: 'POST',
            headers: {
                user_id: data.user_id,
                org_id: data.org_id,
                token: data.token
            }
        }).then(
            function (response) {
                if (response.status !== 200) {
                    return;
                }
                response.json().then(function (json) {
                    if (json.code == 0) {
                        self.setState({"user":{token: data.data }}, function(){
                            Cookies.save('userId', data.user_id, { path: '/' });
                            Cookies.save('orgId', data.org_id, { path: '/' });
                            Cookies.save('token', json.data, { path: '/' });

                            self.setState({ "user": { "token": json.data } })
                            self.setState({ "user": { "user_id": data.user_id } })
                            self.setState({ "user": { "org_id": data.org_id } })
                        })
                        if(printSn){
                            self.getPrinterData(printSn)
                        }else{
                            //获取打印机列表并选择上次的打印机
                            self.getPrinterList({
                                'page': 1,
                                'limit': 10
                            }, function(sn) {
                                self.getPrinterData(sn)
                            });
                        }
                        
                    }
                });
            }
        ).catch(function (err) {
            console.log("错误:" + err);
        });
    }

    //获取打印机列表
    getPrinterList(data, callback) {
        const self = this
        let formData = new FormData();
        formData.append('pageNo', data.page);
        formData.append('pageSize', data.limit);
        //打印机列表
        fetch(mpURL + '/app/printer/queryPage', {
            method: 'POST',
            headers: {
                token: Cookies.load('token')
            },
            body: formData
        }).then(
            function (response) {
                if (response.status !== 200) {
                    return;
                }
                response.json().then(function (json) {
                    if (json.code == 0) {
                        // update 20180904
                        if(json.data.rows.length > 0){
                            let current = 0
                            self.setState({ printer: json.data.rows[current] }, function () {
                                if(typeof callback === 'function'){
                                    callback(json.data.rows[current].printerSn);
                                }
                            });
                        }
                    } else {
                        deli.common.notification.prompt({
                            "type": 'error',
                            "text": json.data,
                            "duration": 1.5
                        },function(data){},function(resp){});
                    }
                });
            }
        ).catch(function (err) {
            deli.common.notification.prompt({
                "type": 'error',
                "text": "网络错误,请重试",
                "duration": 1.5
            },function(data){},function(resp){});
        });
    }

    //获取打印机信息
    getPrinterData(sn){
        const self = this
        //查询打印机信息
        fetch(mpURL + '/app/printer/queryStatus/' + sn, {
            method: 'GET',
            headers: {
                token: Cookies.load('token')
            }
        }).then(
            function (response) {
                if (response.status !== 200) {
                    return;
                }
                response.json().then(function (resp1) {
                    console.log("resp1", resp1)
                    if (resp1.code == 0) {
                        Cookies.save('printer', resp1.data, { path: '/' });
                        self.setState({ printer: resp1.data }, function () {
                            console.log("test2")
                            //查询打印机墨水信息
                            fetch(mpURL + '/app/inkbox/queryDetails/' + self.state.printer.inkboxSn, {
                                method: 'POST',
                                headers: {
                                    token: Cookies.load('token')
                                },
                                body: {}
                            }).then(
                                function (response) {
                                    if (response.status !== 200) {
                                        return;
                                    }
                                    response.json().then(function (resp2) {
                                        if (resp2.code == 0) {
                                            self.setState({ inkbox: resp2.data }, function () {
                                                console.log("test3")
                                            });
                                        }
                                    });
                                }
                            ).catch(function (err) {
                                console.log("错误:" + err);
                            });
                        })
                    } else {

                    }
                });
            }
        ).catch(function (err) {
            console.log("错误:" + err);
        });
    }

    //获取应用信息
    getAppData(timestamp, nonceStr, callback){
        const self = this
        let appData = new FormData();
        appData.append('timestamp', timestamp);
        appData.append('nonceStr', nonceStr);
        //获取应用id
        fetch(mpURL + '/a/auth/config', {
            method: 'POST',
            headers: appData,
            body: appData
        }).then(
            function (response) {
                if (response.status !== 200) {
                    return;
                }
                response.json().then(function (data) {
                    if (data.code === 0) {
                        Cookies.save('appId', data.data.appId, { path: '/' });
                        Cookies.save('sign', data.data.signStr, { path: '/' });
                        /* self.setState({ "user": { token: data.data } }, function () {
                            self.setState({ "user": { "token": data.data } })
                            Cookies.save('token', data.data, { path: '/' });
                            Cookies.save('sn', 'L1000DNW-1B_0000000000000000', { path: '/' });
                        }) */
                        // Cookies.save('token', 'cf8fbcb6-a582-4aaf-b309-0cdb5aaba2f8', { path: '/' });
                        // todo
                        if (typeof callback === 'function'){
                            callback(data)
                        }
                    }
                });
            }
        ).catch(function (err) {
            console.log("错误:" + err);
        });
    }

    render(){
        return(
            <div className="print-index">
                <Printer user={this.state.user} printer={this.state.printer} printer={this.state.printer} inkbox={this.state.inkbox} transPrinter={printer => this.transPrinter(printer)}></Printer>
                <Nav navlist={this.state.navlist} printer={this.state.printer} transPrinter={printer => this.transPrinter(printer)}></Nav>
            </div>
        );
    }
}
PrintIndex.defaultProps = {
    print:{
        printCurrent:0,
        printList:{}
    }
};

export default PrintIndex;