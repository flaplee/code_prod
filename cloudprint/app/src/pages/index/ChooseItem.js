import React, { Component } from 'react'
import { render } from 'react-dom'
import { Redirect } from 'react-router-dom'
import Cookies from 'react-cookies'
import { Group, Boxs, List, Layer, ScrollView } from 'saltui';
import Icon from 'salt-icon';
import './Index.scss';
const { HBox, Box } = Boxs;
import { serverIp, path, baseURL, mpURL, convertURL, timeout, mockURL } from '../../configs/config'

class ChooseItem extends Component{
    constructor(props) {
        super(props);
        this.state = {
            file: props.file,
            index: props.index,
            printLoading: false,
            redirect: { printTask: false },
            printer: {
                sn: (localStorage.getItem('printer') && localStorage.getItem('printer') != "undefined") ? JSON.parse(localStorage.getItem('printer')).printerSn : '',
                name: (localStorage.getItem('printer') && localStorage.getItem('printer') != "undefined") ? JSON.parse(localStorage.getItem('printer')).printerName : '',
                status: (localStorage.getItem('printer') && localStorage.getItem('printer') != "undefined") ? ((JSON.parse(localStorage.getItem('printer')).workStatus == 'error') ? 0 : (JSON.parse(localStorage.getItem('printer')).onlineStatus == '1' ? 1 : 2)) : ''
            },
            printData: {
                'duplexMode': 0,
                'taskSource': (deli.android ? 'ANDROID' : (deli.ios ? 'IOS' : 'WBE')),
                'printDirection': 2,
                'printEndPage': 1,
                'copyCount': 1,
                'printDpi': 600,
                'paperSize': 'A4',
                'printColorMode': 'black',
                'printWhole': 1,
                'printStartPage': 1
            }
        };
    }

    formatDate(time, full, date) {
        let r;
        let t = new Date(time);
        let y = t.getFullYear();
        let m = t.getMonth() + 1;
        let d = t.getDate();
        if (full) {
            r = [t.getHours(), t.getMinutes(), t.getSeconds()];
            if (r[1] < 10) {
                r[1] = '0' + r[1];
            }
            if (r[2] < 10) {
                r[2] = '0' + r[2];
            }
            r = r.join(':');
            r = y + '年' + m + '月' + d + '日' + ' ' + r;
        } else {
            var diff = new Date().Diff('d', t);
            if (diff <= 2 && diff >= -2) {
                r = [t.getHours(), t.getMinutes(), t.getSeconds()];
                if (r[1] < 10) {
                    r[1] = '0' + r[1];
                }
                if (r[2] < 10) {
                    r[2] = '0' + r[2];
                }
                r = r.join(':');
                if (date) {
                    if (diff === 1) {
                        r = '昨天';
                    } else if (diff === 2) {
                        r = '前天';
                    } else if (diff === -1) {
                        r = '明天';
                    } else if (diff === -2) {
                        r = '后天';
                    } else {
                        r = '今天';
                    }
                } else {
                    if (diff === 1) {
                        r = '昨天' + r;
                    } else if (diff === 2) {
                        r = '前天' + r;
                    } else if (diff === -1) {
                        r = '明天' + r;
                    } else if (diff === -2) {
                        r = '后天' + r;
                    } else {
                        r = '今天' + r;
                    }
                }
            } else {
                r = y + '年' + m + '月' + d + '日';
            }
        }
        return r;
    }

    //选择任务打印
    handleItemClick(data) {
        if (data.printTaskType == 'virtual'){
            this.handleVirPrintClick(data)
        }else{
            this.props.transFilerList({ fileItemData: data })
        }
    }

    // 虚拟打印
    handleVirPrintClick(data) {
        const self = this;
        self.setState({ printLoading: true }, function () { })
        let printItems = {};
        const fileList = data.fileList;
        printItems = self.state.printData;
        printItems.taskSource = 'VIRTUAL';
        if (data.downloadUrl) {
            printItems.downloadUrl = data.downloadUrl;
        }
        printItems.printerSn = self.state.printer.sn;
        printItems.fileList = [];
        if (fileList.length == 1){
            printItems.printEndPage = data.fileList[0].totalPage
        }
        for (let i = 0; i < fileList.length; i++) {
            printItems.fileList.push({
                fileSource: fileList[i].fileSource,
                printMd5: fileList[i].printMd5,
                fileSuffix: fileList[i].fileSuffix,
                printPDF: fileList[i].printPDF,
                totalPage: fileList[i].totalPage,
                fileId: fileList[i].fileId,
                printUrl: fileList[i].printUrl,
                fileName: fileList[i].fileName
            })
        }
        //虚拟打印任务设置打印机打印
        fetch(mpURL + '/v1/app/printTask/taskToPrinter/' + data.taskCode +'', {
            method: 'POST',
            timeout: 60000,
            headers: {
                "Content-Type": "application/json",
                "MP_TOKEN": Cookies.load('token')
            },
            body: JSON.stringify(printItems)
        }).then(
            function (response) {
                if (response.status !== 200) {
                    self.setState({ printLoading: false }, function () { })
                    deli.common.notification.toast({
                        "text": "网络错误,请重试",
                        "duration": 1.5
                    }, function (data) { }, function (resp) { });
                    return;
                }
                response.json().then(function (json) {
                    if (json.code == 0) {
                        self.setState({ printLoading: false, redirect: { printTask: true } }, function () { })
                    } else {
                        self.setState({ printLoading: false }, function () { })
                        deli.common.notification.toast({
                            "text": json.msg,
                            "duration": 1.5
                        }, function (data) { }, function (resp) { });
                    }
                });
            }
        ).catch(function (err) {
            self.setState({ printLoading: false }, function () { })
            deli.common.notification.toast({
                "text": "网络错误,请重试",
                "duration": 1.5
            }, function (data) { }, function (resp) { });
        });
    }

    render() {
        const self = this
        let dataIndex = self.state.index
        let dataItem = self.state.file
        //虚拟打印进入打印机任务列表
        if (self.state.redirect.printTask) {
            const sn = self.state.printer.sn
            const status = self.state.printer.status
            const name = self.state.printer.name
            return <Redirect push to={
                { pathname: "/managetask", search: "?sn=" + sn + "&status=" + status + "&name=" + name + "", state: { "sn": sn, "status": status, "name": name } }
            } />;
        }
        return <div key={`page-file-${dataIndex}`} className="task-list-item task-list-item-tap" ref={(input) => { this.fileImgInput = input }} onClick={self.handleItemClick.bind(self, dataItem)}>
            <div className="task-list-item-wrap">
                <HBox vAlign="center">
                    <HBox flex={1}>
                        <Box className="list-item-text-content" flex={1}>
                            <p className="list-item-title t-omit">{dataItem.taskName}</p>
                            <p className="list-item-text t-omit">{self.formatDate(parseInt(dataItem.createTime), true)}</p>
                        </Box>
                    </HBox>
                    <Box>
                        <Icon className="print-list-arrow right" name='direction-right' fill="#ccc" width="7rem" height="3rem" />
                    </Box>
                </HBox>
            </div>
        </div>
    }
}

export default ChooseItem;