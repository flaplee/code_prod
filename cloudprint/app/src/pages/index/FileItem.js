import React, { Component } from 'react'
import { Group, Boxs, List, Layer, ScrollView } from 'saltui';
import Icon from 'salt-icon';
import './Index.scss';

const { HBox, Box } = Boxs;

class FileItem extends Component{

    //打印任务状态
    handleItemClick(item, data) {
        this.props.transFilerList({ layerView: true, layerViewData: item, fileItemData: data })
    }

    render() {
        const self = this
        const statusInfo = {
            'create': {
                status: 'create',
                text: '未开始',
                value: [5, 4, 6]
            },
            'confirm': {
                status: 'confirm',
                text: '等待中',
                value: [3, 4, 6]
            },
            'wating': {
                status: 'wating',
                text: '等待中',
                value: [3, 4, 6]
            },
            'doing': {
                status: 'doing',
                text: '正在打印',
                value: [3, 4, 6]
            },
            'success': {
                status: 'success',
                text: '打印成功',
                value: [1, 2, 6]
            },
            'fail': {
                status: 'fail',
                text: '打印失败',
                value: [1, 2, 6]
            },
            'cancel': {
                status: 'cancel',
                text: '已取消',
                value: [1, 2, 6]
            }
        }
        let dataIndex = self.props.index
        let dataItem = self.props.file
        return <div key={`page-file-${dataIndex}`} className="task-list-item task-list-item-tap" ref={(input) => { this.fileImgInput = input }} onClick={self.handleItemClick.bind(self, statusInfo[dataItem.taskStatus].value, dataItem)}>
            <div className={(statusInfo[dataItem.taskStatus].status == 'fail') ? 'task-list-item-wrap task-list-item-' + statusInfo[dataItem.taskStatus].status +'' : 'task-list-item-wrap'}>
                <HBox vAlign="center">
                    <HBox flex={1}>
                        <Box className="list-item-text-content" flex={1}>
                            <p className="list-item-title t-omit">{dataItem.taskName}</p>
                            <p className="list-item-text t-omit">{dataItem.printerName}</p>
                        </Box>
                    </HBox>
                    <Box className="list-item-text-status">
                        <p className="list-item-status t-omit">{statusInfo[dataItem.taskStatus].text}</p>
                        <i className="list-item-arrow-super"></i>
                    </Box>
                </HBox>
            </div>
        </div>
    }
}

export default FileItem;