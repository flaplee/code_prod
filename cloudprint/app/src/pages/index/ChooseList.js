import React, { Component } from 'react'
import { render } from 'react-dom'
import { Redirect } from 'react-router-dom'
import Cookies from 'react-cookies';
import { Group, Boxs, List, Layer, ScrollView } from 'saltui';
import Icon from 'salt-icon';
import ChooseItem from './ChooseItem'
import { serverIp, path, baseURL, mpURL, convertURL, timeout, mockURL } from '../../configs/config'
import './Index.scss';

const { HBox, Box } = Boxs;

class ChooseList extends Component {

    constructor(props) {
        super(props);
        this.state = {
            printer: props.printer,
            page: props.pages,
            loading: false,
            isEmpty: false,
            noMore: false, 
            force: true,
            printTaskInfo: props.files || [],
            fileList: []
        };

    }

    //监听transFilerList变化状态
    transFilerList(filer) {
        this.props.transFiler(filer)
    }

    //获取列表数据
    getListPage(data) {
        console.log("data~~~~~~~~~~~~", data)
        const self = this
        self.setState({ loading: true });
        deli.common.notification.showPreloader();
        let pages = self.state.fileList;
        let appData = new FormData();
        appData.append('pageNo', data.pageNo);
        appData.append('pageSize', data.pageLimit);
        appData.append('printerSn', data.sn);
        //分页查询打印机任务列表
        fetch(mpURL + '/app/printerTask/queryMyPage', {
            method: 'POST',
            headers: {
                "MP_TOKEN": Cookies.load('token')
            },
            body: appData
        }).then(
            function (response) {
                if (response.status !== 200) {
                    return;
                }
                response.json().then(function (json) {
                    if (json.code == 0) {
                        self.setState({ loading: false });
                        deli.common.notification.hidePreloader();
                        if(data.pageNo == 1 && json.data.total == 0){
                            self.setState({
                                fileList: [],
                                isEmpty: true,
                                loading: false,
                                force: false
                            }, function () {})
                        }else{
                            if (json.data.total > 0){
                                let rows = self.state.fileList
                                self.setState({
                                    fileList: rows.concat(json.data.rows),
                                    force: (json.data.total > (data.pageLimit * data.pageNo)) ? true : false,
                                    noMore: ((data.pageNo == 1) ? false : true),
                                }, function () {})
                            }else{
                                self.setState({
                                    force: false,
                                    noMore: true
                                }, function () { })
                            }
                        }
                    }
                });
            }
        ).catch(function (err) {
            console.log("错误:" + err);
            self.setState({ loading: false });
            deli.common.notification.hidePreloader();
        });
    }

    onLoad() {
        //this.getListPage({ pageNo: this.state.page + 1, pageLimit: 12, sn: this.state.printer.sn })
        /* this.setState({ page: this.state.page + 1, loading: false }, function(){
            console.log("page", this.state.page)
            //this.getListPage({ pageNo: this.state.page, pageLimit: 12, sn: this.state.printer.sn })
        }); */
    }

    render() {
        let fileItems, fileItemEmpty
        if (this.state.isEmpty == true){
            fileItemEmpty = <div className="task-list-empty"><div className="task-list-empty-img"></div><p className="task-list-empty-text">暂无更多打印记录</p></div>
        }else{
            const files = this.state.printTaskInfo
            fileItems = files.map((file, index) => {
                return (
                    <ChooseItem key={index} index={index} file={file} transFilerList={filer => this.transFilerList(filer)} />
                );
            });
        }

        return (
            <ScrollView
                infiniteScroll={this.state.force}
                infiniteScrollOptions={{
                    loading: this.state.loading,
                    onLoad: this.onLoad.bind(this),
                }}
                className={this.state.isEmpty == true ? 'scroll-view-empty' : 'scroll-view-demo'}
            >
                <Group className="list-item">
                    <Group.List lineIndent={15}>
                        {this.state.isEmpty == true ? fileItemEmpty : fileItems}
                        <div className="task-list-more" style={{ display: (this.state.noMore == true) ? 'block' : 'none' }}><p>没有更多了</p></div>
                    </Group.List>
                </Group>
            </ScrollView>
        );
    }
}

export default ChooseList;