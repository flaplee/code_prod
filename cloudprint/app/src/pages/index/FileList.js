import React, { Component } from 'react'
import { render } from 'react-dom'
import { Redirect } from 'react-router-dom'
import Cookies from 'react-cookies';
import { Group, Boxs, List, Layer, ScrollView } from 'saltui';
import Icon from 'salt-icon';
import FileItem from './FileItem'
import { serverIp, path, baseURL, mpURL, convertURL, timeout, mockURL } from '../../configs/config'
import './Index.scss';

const { HBox, Box } = Boxs;

class FileList extends Component {

    constructor(props) {
        super(props);
        this.state = {
            printer: props.printer,
            page: props.pages,
            loading: false,
            isEmpty: props.isEmpty,
            noMore: false, 
            force: true,
            goon: true,
            fileList: []
        };
    }

    //监听transFilerList变化状态
    transFilerList(filer) {
        filer.fileList = this.state.fileList
        this.props.transFiler(filer)
    }

    //获取列表数据
    getListPage(data) {
        const self = this
        self.setState({ loading: false });
        //deli.common.notification.showPreloader();
        let pages = self.state.fileList;
        let appData = new FormData();
        appData.append('pageNo', data.pageNo);
        appData.append('pageSize', data.pageLimit);
        appData.append('printerSn', data.sn);
        //分页查询打印机任务列表
        fetch(mpURL + '/app/printerTask/queryMyPage', {
            method: 'POST',
            timeout: 60000,
            headers: {
                "MP_TOKEN": Cookies.load('token')
            },
            body: appData
        }).then(
            function (response) {
                if (response.status !== 200) {
                    self.setState({ loading: false});
                    //deli.common.notification.hidePreloader();
                    deli.common.notification.toast({
                        "text": '网络错误，请重试',
                        "duration": 2
                    }, function (data) { }, function (resp) { });
                    return;
                }
                response.json().then(function (json) {
                    //deli.common.notification.hidePreloader();
                    if (json.code == 0) {
                        if(data.pageNo == 1 && json.data.total == 0){
                            self.setState({
                                fileList: [],
                                isEmpty: true,
                                loading: false,
                                force: false
                            }, function () {})
                        }else{
                            if (json.data.total > 0){
                                let rows = self.state.fileList
                                self.setState({
                                    loading: false,
                                    page: self.state.page + 1,
                                    fileList: rows.concat(json.data.rows),
                                    force: (json.data.total > (data.pageLimit * data.pageNo)) ? true : false,
                                    noMore: ((data.pageNo == 1) ? false : true),
                                }, function () {})
                            }else{
                                self.setState({
                                    loading: false,
                                    force: false,
                                    noMore: true
                                }, function () {})
                            }
                        }
                    }else{
                        deli.common.notification.toast({
                            "text": json.msg,
                            "duration": 2
                        }, function (data) { }, function (resp) { });
                        self.setState({
                            fileList: [],
                            isEmpty: (data.pageNo == 1) ? true : false,
                            loading: false,
                            force: false
                        }, function () { })
                    }
                });
            }
        ).catch(function (err) {
            self.setState({ loading: false});
            //deli.common.notification.hidePreloader();
            deli.common.notification.toast({
                "text": '网络错误，请重试',
                "duration": 2
            }, function (data) { }, function (resp) { });
        });
    }

    onLoad() {
        if(this.state.force == true){
            const curr = this.state.page;
            this.getListPage({ pageNo: curr, pageLimit: 12, sn: this.state.printer.sn })
        }
    }

    render() {
        let fileItems = this.state.fileList.map((file, index) => {
            return (
                <FileItem key={index} index={index} file={file} transFilerList={filer => this.transFilerList(filer)} />
            );
        });

        if(this.state.isEmpty == true){
            return (
                <div className="task-list-empty"><div className="task-list-empty-img"></div><p className="task-list-empty-text">暂无更多打印记录</p></div>
            );
        }else{
            return (
                <ScrollView
                    infiniteScroll={true}
                    infiniteScrollOptions={{
                        loading: this.state.loading,
                        onLoad: this.onLoad.bind(this),
                    }}
                    className='scroll-view-demo'
                >
                    <Group className="list-item">
                        <Group.List lineIndent={15}>
                            { fileItems }
                            <div className="task-list-more" style={{ display: (this.state.noMore == true) ? 'block' : 'none' }}><p>没有更多了</p></div>
                        </Group.List>
                    </Group>
                </ScrollView>
            );
        }
    }
}

export default FileList;