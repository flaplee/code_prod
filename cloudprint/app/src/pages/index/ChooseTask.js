import React, { Component } from 'react'
import { render } from 'react-dom'
import { Redirect } from 'react-router-dom'
import Cookies from 'react-cookies';
import { Group, Boxs, List, Layer, ScrollView } from 'saltui'
import Icon from 'salt-icon'
import ChooseList from './ChooseList';
import { serverIp, path, baseURL, mpURL, convertURL, timeout, mockURL } from '../../configs/config'
import './Index.scss'

const { HBox, Box } = Boxs;

class ChooseTask extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            page: 1,
            loading: false,
            isEmpty: false,
            refreshing: false,
            printer:{
                sn: (localStorage.getItem('printer') && localStorage.getItem('printer') != "undefined") ? JSON.parse(localStorage.getItem('printer')).printerSn : '',
                name: (localStorage.getItem('printer') && localStorage.getItem('printer') != "undefined") ? JSON.parse(localStorage.getItem('printer')).printerName : '',
                status: (localStorage.getItem('printer') && localStorage.getItem('printer') != "undefined") ? ((JSON.parse(localStorage.getItem('printer')).workStatus == 'error') ? 0 : (JSON.parse(localStorage.getItem('printer')).onlineStatus == '1' ? 1 : 2)) : ''
            },
            fileList: [],
            fileItemData: {},
            printType: 'scan',
            chooseData: JSON.parse(localStorage.getItem('chooseTaskInfo')) || props.location.state.printTaskInfo || {},
            redirect: {
                previewNav: false
            }
        };
    }

    componentWillMount(){
        deli.common.navigation.setTitle({
            "title": "选择任务"
        }, function(data) {}, function(resp) {});

        deli.common.navigation.setRight({
            "text":"",
            "icon": ""
        }, function (data) {}, function (resp) {});

        // 关闭
        deli.common.navigation.close({}, function (data) {
            // 重置
            Cookies.remove('appId');
            Cookies.remove('sign');
            Cookies.remove('userId');
            Cookies.remove('orgId');
            Cookies.remove('token');
            Cookies.remove('admin');
            localStorage.removeItem('printer')
            localStorage.removeItem('printPreviewData')
            localStorage.removeItem('chooseTaskInfo')
        }, function (resp) {});

        deli.common.navigation.goBack({
            "is_go_back": true
        }, function (data) {}, function (resp) {});
    }

    componentDidMount(){}

    onLoad(){}

    //监听Tasker变化状态
    transFiler(filer) {
        this.setState({
            redirect:{
                previewNav: true
            },
            fileItemData: filer.fileItemData
        }, function(){});
    }

    onLoad() {}

    render() {
        if(this.state.redirect.previewNav){
            const file = {
                fileId: this.state.fileItemData.id,
                fileUrl: this.state.fileItemData.pdfUrl,
                fileName: this.state.fileItemData.sourceName,
                fileExt: this.state.fileItemData.fileType
            }
            const sn = this.state.printer.sn
            const name = this.state.printer.name
            const status = this.state.printer.status
            const printType = 'scan'
            const printTaskInfo = this.state.fileItemData
            const fileList = printTaskInfo.fileList
            const fileType = (fileList.length == 1 && fileList[0].fileSuffix == 'other') ? 'other' : ((fileList.length > 1 || (fileList.length == 1 && fileList[0].fileSuffix != 'pdf')) ? 'image' : 'file')
            localStorage.removeItem('printPreviewData')
            localStorage.setItem('printPreviewData', JSON.stringify(fileList))
            localStorage.removeItem('printTaskInfo')
            localStorage.setItem('printTaskInfo', JSON.stringify(printTaskInfo))
            Cookies.save('printPreviewType', fileType, { path: '/' });
            return <Redirect push to={
                { pathname: "/previewindex", search: "?sn=" + sn + "&status=" + status + "&type=" + printType + "", state: { "sn": sn, "fileList": fileList, "fileType": fileType, "printTaskInfo": printTaskInfo } }
            } />;
        }
        let fileListDom;
        if(this.state.isEmpty == true){
            fileListDom = <div className="task-list-empty"><div className="task-list-empty-img"></div><p className="task-list-empty-text">暂无更多打印记录</p></div>;
        }else{
            fileListDom = <ChooseList pages={this.state.page} files={this.state.chooseData} printer={this.state.printer} transFiler={filer => this.transFiler(filer)}></ChooseList>;
        }
        return (
            <div className="print-index-task print-list" id="print-index-task">
                { fileListDom }
            </div>);
    }
}

export default ChooseTask;