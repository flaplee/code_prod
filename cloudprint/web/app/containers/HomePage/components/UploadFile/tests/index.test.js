import React from 'react';
import renderer from 'react-test-renderer';
import { accept, File, regex } from '../index';
import 'jest-styled-components';

it('renders correctly', () => {
  const tree = renderer.create(<File type="file" accept={accept} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('RegExp test result', () => {
  expect(regex.test('wwww.fdstd.com/fdf/pfdn.jpg')).toBe(true);
  expect(regex.test('wwww.fdstd.com/fdf/Doyew.JPG')).toBe(true);

  expect(regex.test('www.fdsf.com/fdf/Doyew.pdf')).toBe(true);

  expect(regex.test('wwww.fdstd.com/fdf/pfdn.xjpg')).toBe(false);
  expect(regex.test('www.fdsf.pdf/fdf/Doyew.rdd')).toBe(false);
  expect(regex.test('wwww.jpg.com/fdf/Doyew.dfd')).toBe(false);
  expect(regex.test('xls.jpg.com/fdf/Doyew.dfd')).toBe(false);

  expect(regex.test('wwww.fdstd.com/fdf/CDGn.mp4')).toBe(false);
});
