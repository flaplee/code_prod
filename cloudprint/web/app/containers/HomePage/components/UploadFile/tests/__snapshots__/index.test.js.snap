// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  display: none;
}

<input
  accept=".jpg,.jpeg,.png,.bmp,.gif,.xls,.xlsx,.doc,.docx,.ppt,.pptx,.txt,.pdf"
  className="c0"
  type="file"
/>
`;
